tests/unit/test_cli.py

```diff
@@ -155,6 +155,35 @@ def test_set_stdin_yaml(cli_runner, vault_with_token):
     assert vault_with_token.db == {"a": {"b": "c"}}
 
 
+def test_set_prompt_value(cli_runner, vault_with_token):
+
+    result = cli_runner.invoke(cli.cli, ["set", "--prompt", "a", "b"])
+
+    assert result.exit_code != 0
+    assert vault_with_token.db == {}
+
+
+def test_set_prompt_value_stdin(cli_runner, vault_with_token):
+
+    result = cli_runner.invoke(cli.cli, ["set", "--prompt", "--stdin", "a"])
+
+    assert result.exit_code != 0
+    assert vault_with_token.db == {}
+
+
+def test_set_prompt(cli_runner, mocker, vault_with_token):
+
+    prompt = mocker.patch("click.prompt")
```

Comment:
This is a valid way to do it, another one would be:

```
result = cli_runner.invoke(cli.cli, ["set", "--stdin", "a"], input="b")
```
(then you could analyse output to search for ```"Please enter value for `a`"```
Which one do you prefer ?

Reply:
I'm not sure I follow you, here. the `prompt` function, if called, is waiting for user input and the test just hang at this step.
So I need to mock `click.prompt`, but again, if I'm mocking it, it won't take the ``input="b"`` as a user-input in this function that won't be called anyway.
And in the output, there's only "Done", because the true "prompt" function isn't called...

So what did you mean there? :/

Reply:
```diff
-    prompt = mocker.patch("click.prompt")
-    prompt.return_value = "b"
-    result = cli_runner.invoke(cli.cli, ["set", "--prompt", "a"])
-    # test for prompt function
-    prompt.assert_called_with("Please enter value for `a`", hide_input=True)
+    result = cli_runner.invoke(cli.cli, ["set", "--prompt", "a"], input="b")
 
     # Correctly stored secret.
     assert result.exit_code == 0
+    assert result.output == "Please enter value for `a`: \nDone\n"
     assert vault_with_token.db == {"a": "b"}
```

Ref http://click.palletsprojects.com/en/5.x/testing/#input-streams

Reply:
"reading user input" is the same (in this case) as "reading stdin". This means if you send something to stdin (using the `input="something` parameter, it will be read as the value the user would enter at the prompt.